libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -728,6 +728,13 @@ int HTTPClient::writeToStream(Stream * stream)
         if(len > 0) {
             ret = writeToStreamDataBlock(stream, len);
 
+            // have we an error?
```

Comment:
Instead of copy-pasting the error clause, can we do something like
````
if (len > 0) {
    ret = writeToStreamDataBlock(stream, len);
} else if ((len == -1 && _client->available() > 0) {
    ret = writeToStreamDataBlock(stream, _client->available() > 0);
}

// Have we an error?
if (ret < 0) {
     return returnError(ret);
}
````

Reply:
Sure thing.  And made a correction at the same time.  Thanks.